wandb/sdk/launch/runner/vertex_runner.py

```diff
@@ -104,114 +96,95 @@ def run(
+            )
+        try:
+            aiplatform.init(
+                project=self.environment.project,
+                location=self.environment.region,
+                staging_bucket=spec_args.get("staging_bucket"),
+                credentials=self.environment.get_credentials(),
+            )
+            job = aiplatform.CustomJob(
+                display_name=launch_project.name,
+                worker_pool_specs=spec_args.get("worker_pool_specs"),
+                base_output_dir=spec_args.get("base_output_dir"),
+                encryption_spec_key_name=spec_args.get("encryption_spec_key_name"),
+                labels=spec_args.get("labels", {}),
+            )
+            execution_kwargs = dict(
+                timeout=run_args.get("timeout"),
+                service_account=run_args.get("service_account"),
+                network=run_args.get("network"),
+                enable_web_access=run_args.get("enable_web_access", False),
+                experiment=run_args.get("experiment"),
+                experiment_run=run_args.get("experiment_run"),
+                tensorboard=run_args.get("tensorboard"),
+                restart_job_on_worker_restart=run_args.get(
+                    "restart_job_on_worker_restart", False
+                ),
+            )
+        # Unclear if there are exceptions that can be thrown where we should
+        # retry instead of erroring. For now, just catch all exceptions and they
+        # go to the UI for the user to interpret.
+        except Exception as e:
+            raise LaunchError("Failed to create Vertex job.") from e
         if synchronous:
-            job.run(service_account=service_account, tensorboard=tensorboard, sync=True)
+            job.run(**execution_kwargs, sync=True)
         else:
-            job.submit(
-                service_account=service_account,
-                tensorboard=tensorboard,
-            )
-
+            job.submit(**execution_kwargs)
         submitted_run = VertexSubmittedRun(job)
-
         while not getattr(job._gca_resource, "name", None):
             # give time for the gcp job object to be created and named, this should only loop a couple times max
             time.sleep(1)
-
-        wandb.termlog(
-            f"{LOG_PREFIX}View your job status and logs at {submitted_run.get_page_link()}."
-        )
         return submitted_run
-
-
-def get_gcp_config(config: str = "default") -> Any:
-    return yaml.safe_load(
-        run_shell(
-            ["gcloud", "config", "configurations", "describe", shlex.quote(config)]
-        )[0]
-    )
-
-
-def exists_on_gcp(image: str, tag: str) -> bool:
-    out, err = run_shell(
-        [
-            "gcloud",
-            "artifacts",
-            "docker",
-            "images",
-            "list",
-            shlex.quote(image),
-            "--include-tags",
-            f"--filter=tags:{shlex.quote(tag)}",
-        ]
-    )
-    return tag in out and "sha256:" in out
```

Comment:
these were already unused and obsoleted by the functionality of the `GcpEnvironment` and `GoogleArtifactRegistry` classes.